src/common_fits.jl

```diff
@@ -20,18 +20,103 @@ function fit_line(xpts, ypts)
+
+    return result.param[2:3], sq_error, Nσ, errors[2:3], fit_curve
+end
+
+"""
+        fit_twofreq_ramsey(xpts, ypts, yvars=[])
+
+Fit one- and two-frequency decaying cosine to Ramsey data, optionally
+taking variances into account, and report which should be used based
+on Akaike's Information Criterion (AIC). This assumes Gaussian
+statistics for each observation.
+"""
+function fit_twofreq_ramsey(xpts, ypts, yvars=[])
+    model(t, p) = ( p[1] * exp(-t ./ p[2]) .* cos(2π * p[3] .*t + p[4]) +
+                    p[5] * exp(-t ./ p[6]) .* cos(2π * p[7] .*t + p[8]) + p[9] )
+    #Use KT estimation to get a guess for the fit
+    freqs,Ts,amps = KT_estimation(ypts-mean(ypts), xpts[2]-xpts[1], 2)
+
+    phases = angle(amps)
+    amps = abs(amps)
+    p_guess = [amps[1], Ts[1], freqs[1], phases[1], amps[2], Ts[2], freqs[2], phases[2], mean(ypts)]
+
+    if isempty(yvars)
+        result2 = curve_fit(model, xpts, ypts, p_guess)
+        sq_err2 = sum((model(xpts, result2.param) - ypts).^2)
+    else
+        result2 = curve_fit(model, xpts, ypts, 1./sqrt(yvars), p_guess)
+        sq_err2 = sum(((model(xpts, result2.param) - ypts).^2)./yvars)
+    end
+    errors2 = estimate_errors(result2)
+    # Compute badness of fit:
+    # Under the null hypothesis (that the model is valid and that the observations
+    # do indeed have Gaussian statistics), the mean squared error is χ² distributed
+    # with `dof2` degrees of freedom. We can quantify badness-of-fit in terms of how
+    # far the observed MSE is from the expected value, in units of σ = 2dof (the expected
+    # standard deviation for the χ² distribution)
+    dof2 = length(xpts)-9
+    Nσ2 = sq_err2/sqrt(2dof2) - dof2/sqrt(2dof2)
+
+    xfine = linspace(xpts[1],xpts[end],1001)
+    fit_curve2 = (xfine, model(xfine, result2.param))
+
+    # Now do 1 freq fit for comparisson TODO: just call fit Ramsey
+    model1(t,p) = p[1]*exp(-t ./ p[2]).*cos(2π*p[3] .* t + p[4]) + p[5]
+    freqs,Ts,amps = KT_estimation(ypts-mean(ypts), xpts[2]-xpts[1], 1)
+
+    p_guess = [abs(amps[1]), Ts[1], freqs[1], angle(amps[1]), mean(ypts)]
+    if isempty(yvars)
+        result1 = curve_fit(model1, xpts, ypts, p_guess)
+        sq_err1 = sum((model1(xpts, result1.param) - ypts).^2)
+    else
+        result1 = curve_fit(model1, xpts, ypts, 1./sqrt(yvars), p_guess)
+        sq_err1 = sum(((model1(xpts, result1.param) - ypts).^2)./yvars)
+    end
+    errors1 = estimate_errors(result1)
+    # Compute badness of fit for this model
+    dof1 = length(xpts)-5
+    Nσ1 = sq_err1/sqrt(2dof1) - dof1/sqrt(2dof1)
+
+    fit_curve1 = (xfine, model1(xfine, result1.param))
+
+    # AIC computation
+    k1 = 5
+    k2 = 9 
+    corr(k,n) = (k+1)*(k+1)/(n-k-2)
+    aicc(e,k,n) = 2 * k + e + corr(k,n)
+
+    aic = aicc(sq_err2,k2,length(xpts)) - aicc(sq_err1,k1,length(xpts))
+
+    return (result1.param, sq_err1, Nσ1, errors1, fit_curve1), (result2.param, sq_err2, Nσ2, errors2, fit_curve2), aic > 0 ? 1 : 2, aic
```

Comment:
If you have the `FitResult` type I defined above, then you could return a two-element `Vector` of FitResults, and use the second argument to be an index into which one AIC determines is better.